src/openrct2/peep/Staff.cpp

```diff
@@ -788,64 +785,63 @@ static uint8_t staff_mechanic_direction_path_rand(Peep* peep, uint8_t pathDirect
             return direction;
     }
     // This will never happen as pathDirections always has a bit set.
-    return peep->PeepDirection;
+    return PeepDirection;
 }
 
 /**
  *
  *  rct2: 0x006C0121
  */
-static uint8_t staff_mechanic_direction_path(Peep* peep, uint8_t validDirections, PathElement* pathElement)
+Direction Staff::MechanicDirectionPath(uint8_t validDirections, PathElement* pathElement)
 {
-    Direction direction = INVALID_DIRECTION;
     uint8_t pathDirections = pathElement->GetEdges();
     pathDirections &= validDirections;
 
     if (pathDirections == 0)
     {
-        return staff_mechanic_direction_surface(peep);
+        return MechanicDirectionSurface();
     }
 
     // Check if this is dead end - i.e. only way out is the reverse direction.
-    pathDirections &= ~(1 << direction_reverse(peep->PeepDirection));
+    pathDirections &= ~(1 << direction_reverse(PeepDirection));
     if (pathDirections == 0)
     {
-        pathDirections |= (1 << direction_reverse(peep->PeepDirection));
+        pathDirections |= (1 << direction_reverse(PeepDirection));
     }
 
-    direction = bitscanforward(pathDirections);
+    Direction direction = bitscanforward(pathDirections);
     pathDirections &= ~(1 << direction);
     if (pathDirections == 0)
     {
-        if (peep->State != PEEP_STATE_ANSWERING && peep->State != PEEP_STATE_HEADING_TO_INSPECTION)
+        if (State != PEEP_STATE_ANSWERING && State != PEEP_STATE_HEADING_TO_INSPECTION)
         {
             return direction;
         }
 
-        if (peep->SubState != 2)
+        if (SubState != 2)
```

Comment:
I'm sure we have enums for substate. Could you use them here?

Reply:
I'm not sure we do for staff sub states

Reply:
Indeed, no enum exists for this yet.